src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx

```diff
@@ -0,0 +1,290 @@
+import React, { FormEvent, useEffect, useMemo, useCallback } from 'react';
+import {
+  GlobeHemisphereEast,
+  Scroll,
+  YoutubeLogo,
+  DiscordLogo,
+  TwitterLogo,
+  TelegramLogo,
+  GithubLogo,
+  FacebookLogo,
+  InstagramLogo,
+} from 'phosphor-react';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  FieldValues,
+  SubmitHandler,
+  useController,
+  useForm,
+  useWatch,
+} from 'react-hook-form';
+import { createColumnHelper, ColumnDef } from '@tanstack/react-table';
+import { LinkType, SocialLinksTableModel } from './types';
+import { formatText } from '~utils/intl';
+import { TableWithMeatballMenuProps } from '~v5/common/TableWithMeatballMenu/types';
+import { LINK_TYPE } from './consts';
+import CellComponent from './partials/CellComponent';
+import { LinkIcon } from './partials';
+
+export const useSocialLinksTableColumns = (
+  name: string,
+): ColumnDef<SocialLinksTableModel, string>[] => {
+  const columnHelper = useMemo(
+    () => createColumnHelper<SocialLinksTableModel>(),
+    [],
+  );
+
+  const columns: ColumnDef<SocialLinksTableModel, string>[] = useMemo(
+    () => [
+      columnHelper.display({
+        id: 'name',
+        header: () => formatText({ id: 'table.row.type' }),
+        cell: ({ row }) => (
+          <CellComponent key={row.id} name={`${name}.${row.index}.name`} />
+        ),
+      }),
+      columnHelper.display({
+        id: 'link',
+        header: () => formatText({ id: 'table.row.url' }),
+        cell: ({ row }) => (
+          <CellComponent key={row.id} name={`${name}.${row.index}.link`} />
+        ),
+      }),
+    ],
+    [columnHelper, name],
+  );
+
+  return columns;
+};
+
+export const useGetTableMenuProps = (
+  { remove },
+  data,
+  toggleChangeSocialLinksModalOn,
+) => {
+  const { field } = useController({
+    name: 'name',
+  });
+
+  return useCallback<
+    TableWithMeatballMenuProps<SocialLinksTableModel>['getMenuProps']
+  >(
+    ({ index }) => ({
+      cardClassName: 'min-w-[9.625rem] whitespace-nowrap',
+      items: [
+        {
+          key: 'edit',
+          onClick: () => {
+            field.onChange({ name: data[index].name, link: data[index].link });
+            toggleChangeSocialLinksModalOn();
+          },
+          label: formatText({ id: 'table.row.edit.link' }),
+          iconName: 'edit-pencil',
+        },
+        {
+          key: 'remove',
+          onClick: () => remove(index),
+          label: formatText({ id: 'table.row.remove' }),
+          iconName: 'trash',
+        },
+      ],
+    }),
+    [data, remove, field, toggleChangeSocialLinksModalOn],
+  );
+};
+
+export const useSocialLinksButtons = (
```

Comment:
I think that it can be just a const. I don't see a reason why should it stay as a hook

Reply:
fixed